test/testPartyAttachments.js

```diff
@@ -0,0 +1,1025 @@
+
+        const responsePut = await httpClient.put({ path: resourceHref, body });
+        assert.equal(responsePut.status, 201);
+
+        // Add attachment
+        debug("PUTting the profile image as attachment");
+        const filesToPut = [
+          {
+            remotefileName: "profile.png",
+            localFilename: "test/images/orange-boy-icon.png",
+            attachmentKey: attachmentKey1,
+            resourceHref,
+          },
+        ];
+        await uploadFilesAndCheck(httpClient, filesToPut);
+
+        const attachmentDownloadUrl =
+          type + "/" + resourceKey + "/attachments/profile.png";
+
+        const filesToPutInvalid = [
+          {
+            remotefileName: "profile2.png",
+            urlToCopy: attachmentDownloadUrl,
+            localFilename: "test/images/little-boy-white.png",
+            attachmentKey: attachmentKey2,
+            resourceHref,
+          }
+        ];
+
+        await uploadFilesAndCheck(httpClient, filesToPut);
+        const href = filesToPut[0].resourceHref;
+        const basePath = href.substring(0, href.lastIndexOf("/"));
+        try {
+          await doPutFiles(httpClient, basePath, filesToPutInvalid);
+          assert.fail('An error is expected !')
+        } catch (err) {
+          assert.equal(err.code, 'streaming.body.read.failed');
+          const partialBodyStr = err.partialBody.toString();
+          assert.equal(partialBodyStr.includes('"status": 400'), true, 'expected status code 400 missing in partial body');
+          assert.equal(partialBodyStr.includes('"code": "upload.and.copy.mix"'), true, 'expected error code missing in partial body');
+        }
+      });
+    });
+
+    describe("customRouteForGet", function () {
+      it("/resource/:key/attachments/:attachmentKey should work", async () => {
+        // this function implicitly tests functions passed to
+        // customRouteForUpload, customRouteForUploadCopy, customRouteForDelete and customRouteForGet
+
+        const body = {
+          type: "person",
+          name: "test user",
+          status: "active",
+        };
+        const resourceKey = uuid.v4();
+        const resourceHref = type + "/" + resourceKey;
+        const attachmentKey = uuid.v4();
+        const localFilename = "test/images/orange-boy-icon.png";
+        const attachmentUrl = `${type}/${resourceKey}/attachments/${attachmentKey}`;
+        const attachmentDownloadUrl =
+          type + "/" + resourceKey + "/attachments/profile.png";
+
+        const responsePut = await httpClient.put({ path: resourceHref, body });
+        assert.equal(responsePut.status, 201);
+
+        // Add attachment
+        debug("PUTting the profile image as attachment");
+        const filesToPut = [
+          {
+            remotefileName: "profile.png",
+            localFilename,
+            attachmentKey,
+            resourceHref,
+          },
+        ];
+        await uploadFilesAndCheck(httpClient, filesToPut);
```

Comment:
No need to check here since we're testing the GET route?

Reply:
I prefer to have the initial upload tested, so in case of an error it is more clear what went wrong.